src/+datools/+variational/ThreeDVar.m

```diff
@@ -66,37 +71,60 @@ function forecast(obj)
         end
 
         function analysis(obj, R, y)
+
+            % A constrained problem like double pendulum will need a
+            % constraint coming in from the object.
+
             dB = obj.BDecomposition;
-            dR = decomposition(R, 'chol');
-            
+
+            if strcmp(class(R), "decomposition")
+                dR = R;
+            else
+                dR = decomposition(R, 'chol');
+            end
+
             xb = obj.State;
             obs = obj.Observation;
 
             t = obj.Model.TimeSpan(end);
 
             H = @(x) obs.observeWithoutError(t, x);
             Hadjoint = @(x) obs.linearization(t, x).';
-            
+
             J = @(x) cost(x, xb, dB, y, dR, H, Hadjoint);
 
-            opts = optimoptions('fmincon','Display','none','SpecifyObjectiveGradient',true, ...
-                'HessianApproximation', {'lbfgs', 50});
-            xa = fmincon(J, xb, [],[],[],[],[],[],[], opts);
+            if strcmp(obj.OptAlg, 'lbfgs')
+                opts = optimoptions('fmincon','Display','none', ...
+                    'SpecifyObjectiveGradient',true, ...
+                    'HessianApproximation', {'lbfgs', 30});
+            else
+
+                hessv = @(Hax, v) dB\v + Hax*(dR\(Hax.'*v));
```

Comment:
I think technically this needs to be Hax' not Hax.', as we want the adjoint not the transpose.